components/supervisor-api/terminal.proto

```diff
@@ -30,6 +30,12 @@ service TerminalService {
         };
     }
 
+    rpc GetOutput(GetOutputRequest) returns (stream GetOutputResponse) {
```

Comment:
I'm very much open to suggestions on the naming here.

Reply:
- What is the contract here relative to a task lifecycle?
- Is it a proper place? This API is about terminals, not tasks. Terminals are not aware about tasks.

Reply:
taks.proto sounds good

Reply:
@filiptronicek Does that make sense to you as well? Let me know if we want to sync on the changes, or if there are any blockers you might be stumbling over. :+1:

Reply:
@geropl I agree with introducing a task service, but if we made the method stream as long as it's active, maybe we should rename it to something like `listen` instead of `get`.

Reply:
> but if we made the method stream as long as it's active, maybe we should rename it to something like listen instead of get.

:+1: The main motivation here is to make more "complete" from the API perspective: IMO we should then be able to direct all calls to this API, and can ditch the `Terminal.Listen` calls alltogether, which would significantly ease our lives in the server handler ([source](https://github.com/gitpod-io/gitpod/blob/8c82d56dbaee2e06d565491c85686dfa36be7b80/components/server/src/workspace/headless-log-service.ts#L321)). WDYT?

Reply:
@akosyakov do you know of a good way to determine the task is completed based on just its id? If we can do that, I'm happy with changing `GetOutput` to `ListenToOutput` to be able to have 2 APIs instead of 3 on the Dashboard.

Reply:
looks at how tasks status is implemented in status.proto

Pudong and Huiwen can support you with supervisor changes 🙏